nsw_da_medical_image/dataset_util/dataset.py

```diff
@@ -134,23 +192,39 @@ def __init__(
         base_path: pathlib.Path,
         videos: list[Video] | None = None,
         planes: list[FocalPlane] | None = None,
+        phases: list[Phase] | None = None,
         transform: typing.Callable[[Image.Image], torch.Tensor] | None = None,
+        *,
+        filter_on_median: bool = False,
```

Comment:
This is the new argument (defaulting to False) to select only the median frame of each phase in the video